lms/djangoapps/courseware/views/views.py

```diff
@@ -354,13 +354,16 @@ def course_info(request, course_id):
             if request.GET.get('upgrade', 'false') == 'true':
                 store_upgrade_cookie = True
                 show_upgrade_notification = True
-            elif upgrade_cookie_name in request.COOKIES and bool(request.COOKIES[upgrade_cookie_name]):
+            elif upgrade_cookie_name in request.COOKIES and course_id in request.COOKIES[upgrade_cookie_name]:
                 show_upgrade_notification = True
 
             if show_upgrade_notification:
                 upgrade_data = VerifiedUpgradeDeadlineDate(course, user)
                 if upgrade_data.is_enabled:
                     upgrade_link = upgrade_data.link
+                else:
```

Comment:
It isn't necessary to add this conditional, and it adds a small amount of additional complexity. I was thinking of it simply as "if you hit the URL with the upgrade parameter, then store that you should see the banner for the next 10 days, assuming that the other upgrade conditions are met".  Functionally, I don't think it matters either way.